js/reference.js

```diff
@@ -850,6 +850,19 @@ var ERMrest = (function(module) {
                     var col = this._ref.columns[i];
                     this._values[i] = col.formatvalue(this._data[col.name], {context:this._ref._context});
                 }
+
+                /*
+                 * use this variable to avoid using computed formatted values in other columns while templating 
+                 */
+                var formattedValues = [];
```

Comment:
The spec says to pre-format all columns before expanding any templates, and to use the pre-formatted values.  This comment seems to contradict that, but I haven't traced all the logic yet to be sure...